### PR TITLE
Dunify

### PR DESCRIPTION
**Pull request by andersfugmann**
Please do not delete the branch after merging, as I want to send a PR upstream also, once we have the license sorted out

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,86 +1,16 @@
...
-clean: $(DEMOS_CLEAN)
-	ocamlbuild -clean
-	rm -f $(BUILD)/$(SHARED_LIB)
-	rm -rf $(GEN)
-	sudo rm -f $(INSTALLED_SHARED_LIB)
-	ocamlfind remove $(LIB_NAME)
-
-.PHONY: default sharedlib lib libinstall install docs clean \
-	$(DEMOS) $(DEMOS_CLEAN)
+.PHONY: build install uninstall run doc
```
Comment (darioteixeira) [inline Makefile:16]: I'd suggest having a `.PHONY` declaration before each rule, instead of this all-encompassing one.

File: README.md [hunk 2]
```diff
@@ -3,15 +3,15 @@
...
-Run `make demos/basic` to run the demo. This demo requires you to
+Run `make run` to run the demo. This demo requires you to
 provide a video file `demos/basic/test.mp4` as input. It also requires
 you to install the `owl` opam package.
 
 Run `make docs` to generate documentation. Open
-`opencv.docdir/index.html` to view. Alternatively, view the
+`_build/default/_doc/_html/index.html` to view. Alternatively, view the
 auto-generated docs online:
 [API](https://calsign.github.io/ocaml-opencv/).
```
Comment (darioteixeira) [inline README.md]: Note that the Makefile target is just `doc`, not `docs`.

File: Makefile [hunk 3]
```diff
@@ -1,86 +1,16 @@
...
-HEADERS=$(GEN)/opencv.h glue.h
-SHARED_LIB=libocamlopencv.so
-SHARED_LIBS_INSTALL_DIR=/usr/lib/
-INSTALLED_SHARED_LIB=$(SHARED_LIBS_INSTALL_DIR)/$(SHARED_LIB)
+run:
+	dune run
 
-MLS=$(GEN)/opencv
-FLAGS=-use-ocamlfind
+doc:
```
Comment (darioteixeira) [inline Makefile:13]: Note that in the README, the Makefile target is declared as `docs`. I do prefer `doc` though, so I'd rather change the README instead.

File: demos/basic/dune [hunk 4]
```diff
@@ -0,0 +1,9 @@
+(executable
+ (name basic)
+ (public_name opencv.basic)
```
Comment (darioteixeira) [inline demos/basic/dune]: Won't giving it a `public_name` mean that the executable will be installed under Opam's `bin` directory? I am not sure if that was the intention with these demos.

File: demos/reuse/dune [hunk 5]
```diff
@@ -0,0 +1,9 @@
+(executable
+ (name reuse)
+ (public_name opencv.reuse)
```
Comment (darioteixeira) [inline demos/reuse/dune]: Same comment as above.

File: dune-project [hunk 6]
```diff
@@ -0,0 +1,17 @@
+(lang dune 2.7)
+(name opencv)
+
+(generate_opam_files true)
+
+(source (github Calsign/ocaml-opencv))
+;(license ISC)
```
Comment (darioteixeira) [inline dune-project]: We now know it's LGPL.

**Review by darioteixeira — COMMENTED** [review]
This is great, thanks! I only found some minor stuff to be addressed.

File: Makefile (hunk 1 above)
Comment (andersfugmann) [reply Makefile:16]: Yes. I was following the pattern already there, so Ill leave it for now.

File: dune-project (hunk 6 above)
Comment (andersfugmann) [reply dune-project]: Yes. I changed that.

File: demos/basic/dune (hunk 4 above)
Comment (andersfugmann) [reply demos/basic/dune]: Indeed. I will remove the public name

File: demos/reuse/dune (hunk 5 above)
Comment (andersfugmann) [reply demos/reuse/dune]: Will fix.

**Review by darioteixeira — APPROVED** [review]

**Pull request merged**